blazorbootstrap/Components/Grid/GridColumn.razor.cs

```diff
@@ -216,51 +224,59 @@ internal RenderFragment HeaderTemplate
-                seq++;
-                builder.OpenElement(seq, "span");
-                seq++;
-                builder.AddAttribute(seq, "class", "me-2");
-                seq++;
-                builder.AddContent(seq, HeaderText);
-                seq++;
-                builder.CloseElement(); // close: span
-
-                if (this.CanSort() && currentSortDirection != SortDirection.None)
+                if (HeaderContent is null)
                 {
+                    if (this.CanSort())
+                    {
+                        seq++;
+                        builder.AddAttribute(seq, "role", "button");
+                        seq++;
+                        builder.AddAttribute(seq, "onclick", async () => await OnSortClickAsync());
+                    }
+
+                    if (this.HeaderTextAlignment != Alignment.None)
+                    {
+                        seq++;
+                        builder.AddAttribute(seq, "class",
+                            BootstrapClassProvider.TextAlignment(this.HeaderTextAlignment));
+                    }
+
                     seq++;
                     builder.OpenElement(seq, "span");
                     seq++;
-                    builder.OpenElement(seq, "i");
+                    builder.AddAttribute(seq, "class", "me-2");
+                    seq++;
+                    builder.AddContent(seq, HeaderText);
                     seq++;
-
-                    var sortIcon = ""; // TODO: Add Parameter for this
-                    if (currentSortDirection == SortDirection.Ascending)
-                        sortIcon = "bi bi-sort-alpha-down";
-                    else if (currentSortDirection == SortDirection.Descending)
-                        sortIcon = "bi bi-sort-alpha-down-alt";
-
-                    builder.AddAttribute(seq, "class", sortIcon);
-                    builder.CloseElement(); // close: i
                     builder.CloseElement(); // close: span
-                }
 
+                    if (this.CanSort() && currentSortDirection != SortDirection.None)
+                    {
+                        seq++;
+                        builder.OpenElement(seq, "span");
+                        seq++;
+                        builder.OpenElement(seq, "i");
+                        seq++;
+
+                        var sortIcon = ""; // TODO: Add Parameter for this
+                        if (currentSortDirection == SortDirection.Ascending)
+                            sortIcon = "bi bi-sort-alpha-down";
+                        else if (currentSortDirection == SortDirection.Descending)
+                            sortIcon = "bi bi-sort-alpha-down-alt";
+
+                        builder.AddAttribute(seq, "class", sortIcon);
+                        builder.CloseElement(); // close: i
+                        builder.CloseElement(); // close: span
+                    }
+                }
+                builder.AddContent(seq, HeaderContent);
```

Comment:
@Lyx52 I think the **else** condition is missing here. Let me know if you think otherwise.

Reply:
@gvreddy04 I was thinking the same thing, but seems that you can use builder.AddContent with null value, and it wont add anything, its similar how ChildContent can also be null even tough its added everytime in CellTemplate
#```csharp
builder.AddContent(102, ChildContent, rowData);
```
But for consistency sake I can also change this.